docs/site/Relations.md

```diff
@@ -0,0 +1,164 @@
+the following are required:
+
+- Use [Dependency Injectition](Dependency-injection.md) to inject an instance of
+  the target repository in the constructor of your source repository class.
+- Declare a property with the factory function type
+  `HasManyRepositoryFactory<TargetModel, typeof TargetModel.prototype.id>` on
+  the source repository class.
+- call the `_createHasManyRepositoryFactoryFor` function in the constructor of
+  the source repository class with the relation name (decorated relation propert
+  on the source model) and target repository instance and assign it the property
+  mentioned above.
+
+The following code snippet shows how it would look like:
+
+{% include code-caption.html
+content="/src/repositories/customer.repository.ts.ts" %}
+
+```ts
+{import} {Order} from '../models/order.model.ts';
+{import} {Customer} from '../models/customer.model.ts';
+{import} {OrderRepository} from './order.repository.ts';
+{import} {DefaultCrudRepository, juggler, HasManyRepositoryFactory} from '@loopback/repository';
+
+class CustomerRepository extends DefaultCrudRepository<
+  Customer,
+  typeof Customer.prototype.id
+> {
+  public orders: HasManyRepositoryFactory<Order, typeof Order.prototype.id>;
+  constructor(
+    @inject('datasources.db') protected db: juggler.DataSource,
+    @repository(OrderRepository) orderRepository: OrderRepository,
+  ) {
+    super(Customer, db);
+    this.orders = this._createHasManyRepositoryFactoryFor(
+      'orders',
+      orderRepository,
+    );
+  }
+}
+```
+
+The following CRUD APIs are now available in the constrained target repository
+factory `orders` for instances of `customerRepository`:
+
+- `create` for creating a target model instance
+- `find` finding target model instance(s)
+- `delete` for deleting target model instance(s)
+- `patch` for patching target model instance(s)
+
+## Using constrained repository in a controller
+
+Once the has many relation has been defined and configured, it can be accessed
+via controller methods to call the underlying constrained repository CRUD APIs
+and exposed in LoopBack's swagger UI once decorated with
+[Route decorators](Routes.md#using-route-decorators-with-controller-methods). It
+will require the value of the foreign key and, depending on the request method,
+a partial value for the target model instance as demonstrated below.
+
+{% include code-caption.html
+content="src/controllers/customer-orders.controller.ts" %}
+
+```ts
+import {post, param, requestBody} from '@loopback/rest';
+import {customerRepository} from '../repositories/customer.repository.ts';
+import {Order} from '../models/order.model.ts';
+
+export class CustomerOrdersController {
+  constructor(
+    @repository(CustomerRepository)
+    protected customerRepository: CustomerRepository,
+  ) {}
+
+  @post('/customers/{id}/order')
+  async createCustomerOrders(
+    @param.path.number('id') customerId: number,
+    @requestBody() orderData: Partial<Order>,
```

Comment:
The type needs to be Order here due to the limitation of `requestBody`. This should be tracked until #1443 is fixed